### DISCUSSION TITLE
I need help

### DISCUSSION THREAD

**Discussion by tomickigrzegorz** [tomickigrzegorz/leaflet-examples#38] [General]
Hi,
I need help, more precisely ideas for examples. It is hard to invent/find possible examples yourself and it is very time-consuming.
If anyone has an idea to add, I will be happy to see the idea.

Thank you in advance for your help.

Greetings

- [x] simple sidebar without plugin

**Comment by tomickigrzegorz** [comment]
Idea:
- marker coordinates stored in indexedDB, adding markers only if they are in map.getBounds()
- simple sidebar without plugin

**Comment by askans** [comment]
Hi

What I would like to know is how to integrate an url with geojson data in a dynamic way, meaning without having the data stored on your own site. An example of such data: https://stadtplan.bonn.de/geojson?OD=4756 from https://opendata.bonn.de/dataset/hallenb%C3%A4der-standorte

For this data the encoding is NOT UTF-8 but Latin-9 if I am not mistaken.

Hope this is a type of idea you are looking for

**Reply by tomickigrzegorz** [reply]
@askans Thanks for the post.
You could expand this sentence because I don't know what you mean "... how to integrate an url with geojson data in a dynamic way, meaning without having the data stored on your own site ..."

As for the site https://opendata.bonn.de/dataset/hallenb%C3%A4der-standorte is in UTF-8 at least as the source of the page indicat ;)

**Reply by askans** [reply]
Sorry to be unclear.
What I mean is that I would like to use just the API call to https://stadtplan.bonn.de/geojson?OD=4756. So whenever I refresh the leaflet webpage the newest data is being used - assuming that the data in https://stadtplan.bonn.de/geojson?OD=4756 is changing from time to time.

**Reply by askans** [reply]
If I am not mistaken UTF-8 does not have problems with special character like "ß" or Umlaute like "ü". In the dataset these characters are misrepresented.

**Reply by tomickigrzegorz** [reply]
You can download data from the API using fetch and L.geojson - https://tomik23.github.io/leaflet-examples/#15.geoJson-simple. Fresh data will be provided if the API changes.

As for utf-8, all European languages fall within its scope. And we use UTF-16 mostly for Asian alphabets - https://github.com/tomik23/reading-time/blob/main/sources/script.js#L35

**Reply by askans** [reply]
Thanks a lot!

**Reply by askans** [reply]
I meant that the data from the source I mentioned to you (City of Bonn, Germany) is not in UTF-8. I did not mean your data.

**Reply by tomickigrzegorz** [reply]
This page is in UTF https://opendata.bonn.de/dataset/batteriesammelbeh%C3%A4lter-standorte and the json https://opendata.bonn.de/api/3/action/package_show?id=6cfb1a3b-d081-42f0-bbbe-2183858cb39c&page=0 on the page is also in UTF but in the javascript object width text with escape characters apparently that's what they need in the app :)
We have a name "Batteriesammelbehälter Standorte" and in json it is encoded "batteriesammelbeh\u00e4lter-standorte" ;)
If you execute this code, you will see that it is displayed correctly :)

```js
const object = {
  namea: "Batteriesammelbehälter Standorte",
  nameb: "batteriesammelbeh\u00e4lter-standorte",
};
console.log(object);
```

They just secured themselves in the data entry application because as you can see from json, html is even stuffed there :)

**Reply by askans** [reply]
Thank you Grzegorz forhelping me to understand!
